Makefile

```diff
@@ -328,9 +328,9 @@ flake8: requirements .flake8
 	touch $(VIRTUALENV_ST2CLIENT_DIR)/bin/activate
 	chmod +x $(VIRTUALENV_ST2CLIENT_DIR)/bin/activate
 
-	$(VIRTUALENV_ST2CLIENT_DIR)/bin/pip install --upgrade "pip==19.3.1"
+	$(VIRTUALENV_ST2CLIENT_DIR)/bin/pip install --upgrade "pip>=19.3.1"
```

Comment:
Any reason we're doing >= here instead of pinning?

Reply:
Good question, the answer is: yes! Pinning these tools to specific versions has caused build failures a few times now, and some of our dependencies are relying on newer and newer versions of tools. So continuing to piecemeal pin versions is just going to cause us more breakage in the future.

Instead of fixing this once a month for the next year, I'm just relaxing the version criteria to allow us to use the most recent versions of those tools. Once we release v3.2 and totally drop support for Python 2.7, we can tighten these requirements back up, because by that point, we will probably stop seeing as much breakage.

Reply:
I'm fine with this. I'll let other comment though, just in case there are any other concerns.

Reply:
Yep, I think @armab might have an opinion on this.

I'll also be adding comments and adjusting all of the pip and setuptools versions to match these requirements as well.

Reply:
:-1: for this approach.

As mentioned in the past, if we want to switch to specific version of pip, setuptools, etc, etc dependencies stackstorm relies on, - that should be done in dedicated PR and across of the other repos: there are a bunch of other places. Inconsistencies in versions result in hard to catch bugs and builds that have less repeatability.

Additionally, relaxing version requirements and relying automatically on latest version of these tools led us to more problems before which are bigger comparing to py2-related issues @blag was mentioning. This is rarely a good approach as it leads to bugs hard to spot due to upstream updates. When upstream updates the behavior, introduces breaking changes and we automatically rely on that, - you newer know what's going on before (suddenly) hitting an issue in production and diving deep into the rabbit hole.

The way that worked better, - update versions periodically _everywhere_. This means periodic maintenance: reviewing what was changed in the new versions of the tools we rely on, test them, try them and then change version interval across all the repositories.

@StackStorm/maintainers WDYT about it?

-----

For wondering, just a couple of issues that we hit in the past:
- https://github.com/StackStorm/st2-packages/pull/549
- https://github.com/StackStorm/st2/issues/4681
- https://github.com/StackStorm/st2/issues/3596
- https://github.com/StackStorm/st2/issues/3595
How pip upgrade should be done:
- https://github.com/StackStorm/st2/pull/4712 (references many other repos)

I bet there are much more examples from @Kami who mostly worked with upgrading upstream versions.

Eg. we can't blindly follow latest version which is just a matter of time to hit us again.

Reply:
I've opened StackStorm/discussions#2 to discuss this further.